x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/form_validation/downsample_interval.test.ts

```diff
@@ -0,0 +1,130 @@
+
+    await act(async () => {
+      testBed = await setupValidationTestBed(httpSetup);
+    });
+
+    const { component } = testBed;
+    component.update();
+    ({ actions } = testBed);
+    await actions.setPolicyName('mypolicy');
+  });
+
+  [
+    {
+      name: `doesn't allow empty interval`,
+      value: '',
+      error: [i18nTexts.editPolicy.errors.numberRequired],
+    },
+    {
+      name: `doesn't allow 0 for interval`,
+      value: '0',
+      error: [i18nTexts.editPolicy.errors.numberGreatThan0Required],
+    },
+    {
+      name: `doesn't allow -1 for interval`,
+      value: '-1',
+      error: [i18nTexts.editPolicy.errors.numberGreatThan0Required],
+    },
+    {
+      name: `doesn't allow decimals for timing (with dot)`,
+      value: '5.5',
+      error: [i18nTexts.editPolicy.errors.integerRequired],
+    },
+    {
+      name: `doesn't allow decimals for timing (with comma)`,
+      value: '5,5',
+      error: [i18nTexts.editPolicy.errors.integerRequired],
+    },
+  ].forEach((testConfig: { name: string; value: string; error: string[] }) => {
+    (['hot', 'warm', 'cold'] as PhaseWithDownsample[]).forEach((phase: PhaseWithDownsample) => {
+      const { name, value, error } = testConfig;
+      test(`${phase}: ${name}`, async () => {
+        if (phase !== 'hot') {
+          await actions.togglePhase(phase);
+        }
+
+        await actions[phase].downsample.toggle();
+
+        // 1. We first set as dummy value to have a starting min_age value
+        await actions[phase].downsample.setDownsampleInterval('111');
+        // 2. At this point we are sure there will be a change of value and that any validation
+        // will be displayed under the field.
+        await actions[phase].downsample.setDownsampleInterval(value);
+
+        actions.errors.waitForValidation();
+
+        actions.errors.expectMessages(error);
+      });
+    });
+  });
+
+  test('should validate an interval is greater or multiple than previous phase interval', async () => {
+    await actions.togglePhase('warm');
+    await actions.togglePhase('cold');
+
+    await actions.hot.downsample.toggle();
+    await actions.hot.downsample.setDownsampleInterval('60', 'm');
+
+    await actions.warm.downsample.toggle();
+    await actions.warm.downsample.setDownsampleInterval('1', 'h');
+
+    actions.errors.waitForValidation();
+    actions.errors.expectMessages(
+      ['Must be greater than and a multiple of the hot phase value (60m)'],
+      'warm'
+    );
+
+    await actions.cold.downsample.toggle();
+    await actions.cold.downsample.setDownsampleInterval('90', 'm');
+    actions.errors.waitForValidation();
+    actions.errors.expectMessages(
+      ['Must be greater than and a multiple of the warm phase value (1h)'],
+      'cold'
+    );
+
+    // disable warm phase;
+    await actions.togglePhase('warm');
+    // TODO: there is a bug that disabling a phase doesn't trigger downsample validation in other phases,
```

Comment:
Is there a separate issue open to address this? Will the validation work as expected on save?

Reply:
> Will the validation work as expected on saving?

Yes! the form state stays in the error state until the validation is triggered again.
More details on the case here: https://github.com/elastic/kibana/pull/138748#issuecomment-1243637515

> note: there is the following edge case:
> Have the warm downsample action enabled and set 1d
> Have the cold downsample action enabled and set 1d -> see the error
> Disable the warm step completely -> error on cold downsample action is still there > until triggering change or saving.
> To fix this, we need to validate downsample actions when steps are toggled. But I thought this is an overkill for this edge case.


> Is there a separate issue open to address this?

I wanted to leave a TODO comment to point this out as a known bug/edge case but didn't think this was worth an issue

Reply:
I'll see if I can fix this with low effort: https://github.com/elastic/kibana/pull/140628